app/src/main/java/com/duckduckgo/app/global/view/StringHtmlExtension.kt

```diff
@@ -35,3 +35,16 @@ private fun htmlDrawable(context: Context, resource: Int): Drawable {
     drawable.setBounds(0, 0, drawable.intrinsicWidth, drawable.intrinsicHeight)
     return drawable
 }
+
+private const val HTTPS_PREFIX = "https://"
+private const val WWW_PREFIX = "https://www."
+private const val WWW_SUFFIX = "/"
+
+fun String.website(): String {
+    return this.takeIf { it.startsWith(WWW_PREFIX, ignoreCase = true) && it.endsWith(WWW_SUFFIX, ignoreCase = true) }
```

Comment:
I'm confused what this function is doing. It drops `https://www` from the start, and a trailing `/` from the end, but only if both are present? `&&`

what about `https://www.example.com`, would we want to drop the prefix from that too even though it doesn't have a trailing slash? 

(or maybe i'm not understanding the function fully; i find it hard to parse)

Reply:
Agree naming could be better. This is formatting the `origin` that we receive from `onGeolocationPermissionsShowPrompt`. The format is always the same, `https://www` (http is no longer allowed) and `/` at the end.

How about something like `String.websiteFromGeoLocationsApiOrigin()`?

Reply:
Thanks for explaining. I think it's partly the _fancy_ kotlin that's throwing me here. You could take advantage of the fact that the suffix isn't case-sensitive, and write it in a more imperative way that i think is easier to understand e.g.
```
fun String.websiteFromGeoLocationsApiOrigin(): String {
    var extracted = this.removeSuffix(WWW_SUFFIX)

    if (extracted.startsWith(WWW_PREFIX, ignoreCase = true)) {
        extracted = extracted.drop(WWW_PREFIX.length)
    }

    return extracted
}
```

But two things

1. Might just be a personal preference thing; will leave it with you
1. I haven't tested that code! 😅